### ISSUE TITLE
Allow CaseInsensitiveArray to be traversed

### ISSUE THREAD

**Issue by pbowyer** [stripe/stripe-php#735]
Stripe's exceptions return a lot of extra information about the HTTP request, which is awesome when debugging problems.

The HTTP Headers are held in a `Stripe\Util\CaseInsensitiveArray` collection. This makes it easy to get individual headers, but there's no way to get all of them, either via a `->all()` method or by iterating:
```
foreach ($exception->getHttpHeaders() as $k => $v) {
   // This is never entered
}
```

When handling the exception I would like to record all headers, so getting them would be very handy.

**Comment by ob-stripe** [comment]
Hi @pbowyer, thanks for the suggestion! We just released stripe-php 7.1.0. `CaseInsensitiveArray` is now countable and traversable.

**Comment by pbowyer** [comment]
Brilliant, thank you!
